doomed_patient/doomed_patient/NPC.cpp

```diff
@@ -10,18 +10,107 @@ NPC::NPC(PatientGame* game, VectorXY startCoordinates, Texture * sprite)
 	int random = rand() % 4;
 	movementDirection = static_cast<Directions::Direction>(random);
 }
+
+
 void NPC::move(Directions::Direction direction)
 {
 	centre += Directions::getDirectionVector(direction) * speed;
 }
+
+
 void NPC::update(LevelCell::Passable blocked)
 {
+	// Makes sure NPC's grid position is up to date
+	updateCurrentCell();
+
+	// Update direction if necessary
+	updateDirection();
+
+	move(movementDirection);
+}
+
+// Call NPC behaviour code/methods that change NPC movementDirection from this method!
+void NPC::changeDirection()
```

Comment:
This method is called from ```NPC::updateDirection()```. Any methods you create for your NPC behaviour that change their ```movementDirection``` should be called from this method, to ensure that movement direction change is only carried out after ```NPC::updateDirection()``` has checked the NPC's position